### DISCUSSION TITLE
Missing server configurations

### DISCUSSION THREAD

**Discussion by asmodeus812** [mason-org/mason.nvim#724] [Q&A]
Hi i was just looking through the mason-lspconfig package and compared to the nvim-lspinstaller i notice that a lot of the server default options / config files are not even there - for example the old lsp installer had a default options config for jdtls, which computed the jdtls command and we can use that instead of having to configure one manually. For example: https://github.dev/williamboman/nvim-lsp-installer/blob/17e0bfa5f2c8854d1636fcd036dc8284db136baa/lua/nvim-lsp-installer/servers/jdtls/init.lua. Same issues are observed for other servers, but the old installer had a nice default options exposed out of the box, the new one seems to only glue the server executable name in the cmd and that is it. What am i missing ?

**Comment by williamboman** [comment]
Hello! So `jdtls` recently started shipping a `jdtls` executable script that handles reading configuration options and whatnot. `lspconfig` [uses this script](https://github.com/williamboman/nvim-lspconfig/blob/f938e3b68675770471d5d03ffb595abdc9410b31/lua/lspconfig/server_configurations/jdtls.lua#L93-L100) and passes configuration options accordingly. By having mason-lspconfig not involve itself more than absolutely necessary the boundaries between the triad of plugins (mason, mason-lspconfig, lspconfig) becomes a bit clearer. If you think some configuration items are missing for `jdtls` it's best reported to lspconfig (preferably in the form of a PR)!
